pkgs/by-name/wr/wrangler/package.nix

```diff
@@ -0,0 +1,107 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  nodejs,
+  pnpm_9,
+  # cpp is required for building workerd.
+  # workerd is used for some wrangler subcommands.
+  # non-linux platforms may experience errors
+  # on certain subcommands due to this issue.
+  autoPatchelfHook,
+  llvmPackages,
+  musl,
+  xorg,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wrangler";
+  version = "3.62.0";
+
+  src = fetchFromGitHub {
+    owner = "cloudflare";
+    repo = "workers-sdk";
+    rev = "wrangler@${finalAttrs.version}";
+    hash = "sha256-/4iIkvSn85fkRggmIha2kRlW0MEwvzy0ZAmIb8+LpZQ=";
+  };
+
+  buildInputs =
+    # cpp is required for building workerd.
+    # workerd is used for some wrangler subcommands.
+    # non-linux platforms may experience errors
+    # on certain subcommands due to this issue.
+    lib.optional stdenv.isLinux
+    [
+      autoPatchelfHook
+      llvmPackages.libcxx
+      llvmPackages.libunwind
+      musl
+      xorg.libX11
+    ]
+
+  nativeBuildInputs =
+    [
+      makeWrapper
+      nodejs
+      pnpm_9.configHook
+    ]
+    ++
+    # cpp is required for building workerd.
+    # workerd is used for some wrangler subcommands.
+    # non-linux platforms may experience errors
+    # on certain subcommands due to this issue.
+    lib.optional stdenv.isLinux autoPatchelfHook;
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-aTTaiGXm1WYwmy+ljUC9yO3qtvN20SA+24T83dWYrI0=";
+  };
```

Comment:
I think it'd be better to put this near the `src`, as the hashes will be near each other and it will be a bit harder to miss if in a future update this hash won't be updated. Also, it makes the diffs a bit smaller this way when the hashes are near each other.

Reply:
on it